chore!: upgrade Django version to 4.2 (LTS)

iamsobanjaved opened the pull request:
Upgrade to Django 4.2

Issue: #32833 

This is final django42 upgrade PR for production.

- [x] sandbox created
- [x] ecommerce checkout working
- [x] import/export and few celery tasks tested.
- [x] makemigration does not bring new migration.
- [x] migrate does not run any hidden migration. 
- [x] sha1 value updated to sha256 in settings.py 


Original PR https://github.com/openedx/edx-platform/pull/33211

iamsobanjaved committed:
chore!: upgrade Django version to 4.2 (LTS)

This reverts commit 6f32d0e6f1596445822a7e964ffd6928d0d5415b.

awais786 commented:
https://github.com/openedx/edx-platform/pull/33969 
Last attempt failed due to one xblock version issue.

iamsobanjaved committed:
feat: Upgrade Python dependency Django

chore: upgrade Django to latest patch version

Commit generated by workflow `openedx/edx-platform/.github/workflows/upgrade-one-python-dependency.yml@refs/heads/master`

iamsobanjaved committed:
Merge pull request #34119 from openedx/iamsobanjaved/upgrade-Django-67d53dc

feat: Upgrade Python dependency Django

awais786 committed:
Merge branch 'master' into iamsobanjaved/django-42-lts

awais786 committed:
Merge branch 'master' into iamsobanjaved/django-42-lts
